### PR TITLE
Add beta channel to /get-dart

### PR DESCRIPTION
**Pull request by mit-mit**
* Update get-dart with beta channel description
* Add beta channel SDK instructions
* Clean up OS specific installation pages to always recommend stable SDKs first

Partial fix for https://github.com/dart-lang/site-www/issues/2326

### CODE REVIEW

File: src/get-dart.md [hunk 1]
```diff
@@ -49,28 +49,37 @@ so it's easy to find (but not more tempting than package managers).
...
+* **stable** channel: **stable releases**, updated roughly every three months;
   currently `[calculating]`{:.editor-build-rev-stable}.
-* **dev** channel: **prereleases**, usually updated 1/week;
+  
+  Stable releases are suitable for production use.
+  
+* **beta** channel: **candidate releases**, usually updated every month;
+  currently `[calculating]`{:.editor-build-rev-beta}.
+  
+  Beta channel builds are candidate builds for the stable channel. We recommend
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply src/get-dart.md:62]: I'm just going to change "are candidate builds" to "are preview builds"

**Comment by mit-mit** [comment]
@athomas I discussed this with @kwalrath and we agreed to really simplify this page by make it focused on installing and upgrading stable SDKs, and then pointing at the SDK archive for beta and dev channels.

To not use the content for brew|choco|apt-get for beta and dev channels for those that should prefer that, I moved that to https://github.com/dart-lang/sdk/wiki/Installing-beta-and-dev-releases-with-brew,-choco,-and-apt-get. Can you help with getting the apt-get steps there added?

File: src/tools/sdk/_windows.md [hunk 2]
```diff
@@ -1,44 +1,19 @@
...
 
-To install a **dev** release, run this command:
-
-```terminal
-C:\> choco install dart-sdk --pre
-```
+### Upgrade
 
 To **upgrade** the Dart SDK, run this command
 (add `--pre` to upgrade the dev release):
```
Comment (kwalrath) [inline src/tools/sdk/_windows.md:13]: Delete this line about the dev release

**Comment by mit-mit** [comment]
Staged: https://mit-staging.web.app/get-dart

File: src/tools/sdk/_mac.md [hunk 3]
```diff
@@ -39,28 +18,14 @@ To upgrade when a new release of Dart is available run:
...
 ### Switch release
 
 To switch between locally installed dart releases run
 `brew switch dart <version>`. Examples:
 
 ```terminal
-$ brew switch dart 1.24.3
+$ brew switch dart 2.
 $ brew switch dart 2.1.0
 ```
```
Comment (kwalrath) [inline src/tools/sdk/_mac.md:31]: Delete the last two lines in this file? (They mention the dev version.)

File: src/get-dart.md [hunk 4]
```diff
@@ -47,32 +48,42 @@ so it's easy to find (but not more tempting than package managers).
...
+**Stable** channel releases of the Dart SDK have `x.y.z` version strings like
+`1.24.3` and `2.1.0`. They consist of dot-separated integers, with no hyphens or
+letters, where `x` is the major version, `y` is the minor version, and `z` is
+the patch version.
 
-**Dev** channel releases of the Dart SDK (prereleases)
-have additional characters, starting with a hyphen (`-`).
-For example, Dart 2 prereleases have version numbers starting with
-`2.0.0-dev` such as `2.0.0-dev.69.5`.
+**Beta** and **Dev** channel releases of the Dart SDK (non-stable releases) have
```
Comment (kwalrath) [inline src/get-dart.md:86]:
```suggestion
**Beta** and **dev** channel releases of the Dart SDK (non-stable releases) have
```

File: src/get-dart.md [hunk 5]
```diff
@@ -47,32 +48,42 @@ so it's easy to find (but not more tempting than package managers).
...
+`x.y.z-a.b.<beta|dev>` versions like `2.8.0-20.11.beta`. The part before the
+hyphen follows the stable version scheme, `a` and `b` after the hyphen is the
+pre-release and pre-release patch versions, and `beta` or `dev` is the channel.
```
Comment (kwalrath) [inline src/get-dart.md:89]:
```suggestion
prerelease and prerelease patch versions, and `beta` or `dev` is the channel.
```

File: src/get-dart.md [hunk 6]
```diff
@@ -47,32 +48,42 @@ so it's easy to find (but not more tempting than package managers).
...
+letters, where `x` is the major version, `y` is the minor version, and `z` is
+the patch version.
 
-**Dev** channel releases of the Dart SDK (prereleases)
-have additional characters, starting with a hyphen (`-`).
-For example, Dart 2 prereleases have version numbers starting with
-`2.0.0-dev` such as `2.0.0-dev.69.5`.
+**Beta** and **Dev** channel releases of the Dart SDK (non-stable releases) have
+`x.y.z-a.b.<beta|dev>` versions like `2.8.0-20.11.beta`. The part before the
+hyphen follows the stable version scheme, `a` and `b` after the hyphen is the
```
Comment (kwalrath) [inline src/get-dart.md:88]:
```suggestion
hyphen follows the stable version scheme, `a` and `b` after the hyphen are the
```

**Comment by kwalrath** [comment]
LGTM once you address my feedback.

**Pull request merged**